src/classes/HiFiMixerSession.ts

```diff
@@ -1317,4 +1398,104 @@ export class HiFiMixerSession {
+        [this.userDataStreamingScope, secondarySession.userDataStreamingScope] = [secondarySession.userDataStreamingScope, this.userDataStreamingScope];
+        [this.onUserDataUpdated, secondarySession.onUserDataUpdated] = [secondarySession.onUserDataUpdated, this.onUserDataUpdated];
+        [this.onUsersDisconnected, secondarySession.onUsersDisconnected] = [secondarySession.onUsersDisconnected, this.onUsersDisconnected];
+        [this.onMuteChanged, secondarySession.onMuteChanged] = [secondarySession.onMuteChanged, this.onMuteChanged];
+        [this.onTransitionRequested, secondarySession.onTransitionRequested] = [secondarySession.onTransitionRequested, this.onTransitionRequested];
+        [this.onTransition, secondarySession.onTransition] = [secondarySession.onTransition, this.onTransition];
+        [this.onReadyForTransition, secondarySession.onReadyForTransition] = [secondarySession.onReadyForTransition, this.onReadyForTransition];
+        [this._lastSuccessfulInputAudioMutedValue, secondarySession._lastSuccessfulInputAudioMutedValue] = [secondarySession._lastSuccessfulInputAudioMutedValue, this._lastSuccessfulInputAudioMutedValue];
+
+        this._disableReconnect = true;
+        secondarySession._disableReconnect = false;
+
+        HiFiLogger.debug("Setting new primary/old secondary session mute to communicator mute value.");
+        await secondarySession._trySetInputAudioMuted(secondarySession._lastSuccessfulInputAudioMutedValue, false);     
+        HiFiLogger.debug("Muting old primary/new secondary mixer session.");
+        await this._trySetInputAudioMuted(true, false);
+
+        // send deleted peer notifications for any peers that didn't make it over to the new mixer
+        // and send new peer data notifications for all peers that are on the new mixer.
+        let allDeletedUserData: Array<ReceivedHiFiAudioAPIData> = [];
+        let allNewUserData: Array<ReceivedHiFiAudioAPIData> = [];
+            
+
+        let oldPeers: any = {};
+        for (const mixerPeerKey of Object.keys(this._mixerPeerKeyToStateCacheDict)) {
+            oldPeers[this._mixerPeerKeyToStateCacheDict[mixerPeerKey].hashedVisitID] = this._mixerPeerKeyToStateCacheDict[mixerPeerKey];
+        }
+        for (const mixerPeerKey of Object.keys(secondarySession._mixerPeerKeyToStateCacheDict)) {
+            delete oldPeers[secondarySession._mixerPeerKeyToStateCacheDict[mixerPeerKey].hashedVisitID];
+            allNewUserData.push(secondarySession._mixerPeerKeyToStateCacheDict[mixerPeerKey]);
+        }
+        
+        for (const hashedVisitID of Object.keys(oldPeers)) {
+            let deletedUserData = new ReceivedHiFiAudioAPIData({
+                hashedVisitID: hashedVisitID
+            });
+            if (oldPeers[hashedVisitID].providedUserID) {
+                deletedUserData.providedUserID = oldPeers[hashedVisitID].providedUserID;
+            }
```

Comment:
Similarly this would be faster:
```
let username = oldPeers[hashedVisitID].providedUserID;
if (username) {
    deletedUserData.providedUserID = username;
}
```

Reply:
fixing